test/behavior/floatop.zig

```diff
@@ -114,3 +139,279 @@ test "more @sqrt f16 tests" {
     try expect(math.isNan(@sqrt(@as(f16, -1.0))));
     try expect(math.isNan(@sqrt(@as(f16, math.nan(f16)))));
 }
+
+test "@sin" {
+    comptime try testSin();
+    try testSin();
+}
+
+fn testSin() !void {
+    inline for ([_]type{ f16, f32, f64 }) |ty| {
+        const eps = epsForType(ty);
+        try expect(@sin(@as(ty, 0)) == 0);
+        const pi_over_2 = @as(ty, std.math.pi / 2);
```

Comment:
Note that the compile error from the CI is coming *only* from stage1. Stage2 has no problem with this line. I'm OK with it if you just want to disable the entire `@sin` test for stage1 since it gets incorrect results.

```zig
if (builtin.zig_backend == .stage1) {
    // stage1 emits an incorrect compile error and gets wrong float results for some of the test cases
    return error.SkipZigTest;
}
```

Reply:
Skipped the failing parts of the tests for now 👍